Editor/Window/ConnectToFleetInput.cs

```diff
@@ -49,14 +49,15 @@ public ConnectToFleetInput(VisualElement container, StateManager stateManager)
             {
                 await UpdateFleetMenu();
                 
-                var fleet = _fleetAttributes.FirstOrDefault(fleet => fleet.Name == _stateManager.AnywhereFleetName);
-                if (fleet == null)
+                if (_fleetAttributes.Count <= 0)
                 {
                     _fleetState = FleetStatus.NotCreated;
                 }
                 else
                 {
-                    _fleetState = FleetStatus.Selected;
+                    var fleet = _fleetAttributes.FirstOrDefault(fleet => fleet.Name == _stateManager.AnywhereFleetName);
+
+                    _fleetState = fleet == null ? FleetStatus.Selecting : FleetStatus.Selected;
```

Comment:
If we don't have their already selected fleet, but the list is still more than 0, they can select a fleet that they can use.

Reply:
A similar check happens in line 202-205. 

Can you write a small method called SetFleetState() that just does this for you and reuse the method instead of 202-25. This will help keep the conditions constant and avoid edge case bugs.

Reply:
Fixed